generators/pkg/generator/generate.go

```diff
@@ -144,3 +155,163 @@ func GeneratePipeline(github *GitHub) (*v1beta1.Pipeline, error) {
 
 	return pipeline, nil
 }
+
+// Generate the trigger with the given generated pipeline
+func GenerateTrigger(p *v1beta1.Pipeline) *Trigger {
+	if p.Namespace == "" {
+		p.Namespace = "default"
+	}
+
+	// create pipelinerun in the triggertemplate
+	pr := &v1beta1.PipelineRun{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "PipelineRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    p.Namespace,
+			Name:         p.Name,
+			GenerateName: p.Name + "-run-",
+			Labels:       p.Labels,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineRef: &v1beta1.PipelineRef{
+				Name: p.Name,
+			},
+			Params: []v1beta1.Param{
+				{
+					Name: "gitrepositoryurl",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "$(params.gitrepositoryurl)",
+					},
+				},
+				{
+					Name: "gitrevision",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: "$(params.gitrevision)",
+					},
+				},
+			},
+			Workspaces: []v1beta1.WorkspaceBinding{
+				{
+					Name: p.Spec.Workspaces[0].Name,
+					VolumeClaimTemplate: &corev1.PersistentVolumeClaim{
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes: []corev1.PersistentVolumeAccessMode{
+								corev1.ReadWriteOnce,
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"storage": *resource.NewQuantity(1*1024*1024*1024, resource.BinarySI),
```

Comment:
Fine for now, but we should think about how we can make this configurable in the future.